cpp/src/sort/is_sorted.cu

```diff
@@ -37,13 +38,15 @@ auto is_sorted(cudf::table_view const& in,
 {
   // 0-table_view, 1-column_order, 2-null_precedence, 3-validity_columns
   auto flattened = structs::detail::flatten_nested_columns(in, column_order, null_precedence);
-  auto in_d      = table_device_view::create(std::get<0>(flattened), stream);
-  rmm::device_vector<order> d_column_order(std::get<1>(flattened));
-  rmm::device_vector<null_order> const d_null_precedence =
-    (has_nulls) ? rmm::device_vector<null_order>{std::get<2>(flattened)}
-                : rmm::device_vector<null_order>{};
+
+  auto in_d           = table_device_view::create(std::get<0>(flattened), stream);
+  auto d_column_order = make_device_uvector_async(std::get<1>(flattened), stream);
+
   auto ineq_op = row_lexicographic_comparator<has_nulls>(
-    *in_d, *in_d, d_column_order.data().get(), d_null_precedence.data().get());
+    *in_d,
+    *in_d,
+    d_column_order.data(),
+    has_nulls ? make_device_uvector_async(std::get<2>(flattened), stream).data() : nullptr);
```

Comment:
The previous one was clearer.

Reply:
The problem is the previous required creating an empty device_vector so it could simply pass nullptr to this function. It seemed a waste to create an empty vector just for that. But I can do that if that is preferred.

Reply:
It'll get optimized out. I agree w/ Devavret.

Reply:
I don't think an empty uvector allocates space even when not optimised out. [rmm](https://github.com/rapidsai/rmm/blob/befea87f1dc7dc5fd008039057752579ed1775d3/include/rmm/device_buffer.hpp#L409)